### PR TITLE
crimson/common: use seastar API level 5

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
the change on the seastar side is being review at https://groups.google.com/g/seastar-dev/c/HdkKJf6d4uk/m/2Nr1dRV8AgAJ

File: src/crimson/common/errorator.h [hunk 1]
```diff
@@ -835,23 +835,44 @@ struct errorator {
...
+      typename stored_to_future<typename vanilla_futurize::value_type>::type;
 
     template <class Func, class... Args>
     static type apply(Func&& func, std::tuple<Args...>&& args) {
       return vanilla_futurize::apply(std::forward<Func>(func),
                                      std::forward<std::tuple<Args...>>(args));
     }
 
+    template <class Func, class... Args>
+    static type invoke(Func&& func, Args&&... args) {
```
Comment (rzarzynski) [inline src/crimson/common/errorator.h:858]: Any chance to drop the `std::tuple`-taking `apply()` above?

**Review by rzarzynski — APPROVED** [review]
Looks sane. :-)

File: src/crimson/common/errorator.h (hunk 1 above)
Comment (tchaikov) [reply src/crimson/common/errorator.h:858]: to drop the `static type apply(Func&& func, std::tuple<Args...>&& args)` method? i wanted to play safe. will try.

Comment (tchaikov) [reply src/crimson/common/errorator.h:852]: @rzarzynski  yeah. seems `apply()` is not used anymore. repushed.

Comment (rzarzynski) [reply src/crimson/common/errorator.h:852]: :+1:

**Pull request merged**